### DISCUSSION TITLE
MiEl HVAC only shows MiElHVAC: {} from Feather Huzzah with MSZ-GL06NA, why no communication / how to debug?

### DISCUSSION THREAD

**Discussion by markfickett** [arendst/Tasmota#19815] [Q&A]
I'd like to set up HomeAssistant control of my Mitsubishi heat pump, via Tasmota and MQTT. I'm using [isaiahchia's post](https://isaiahchia.com/2022/06/16/integrating-mitsubishi-air-con-hvac-with-home-assistant-via-nodemcutasmota/) as a reference.

I compiled a custom Tasmota build with `#define USE_MIEL_HVAC`, flashed `build_output/firmware/tasmota.bin` to a Feather Huzzah ESP8266, and configured it as a `Generic (18)` with `Tx` = `MiEl HVAC Tx` and `Rx` = `MiEl HVAC Rx`.

When I connect it up to the CN105 port (via a pigtail from AliBaba) of my MSZ-GL06NA, I see the Tasmota device, but in the MQTT output I just see `"MiElHVAC": {}` in the sensor topic. I would expect that to have mode, temperature, etc information if it were communicating. In the Tasmota console, I additionally see a bunch of line saying 'set to 2400' 'set to 9600' repeating, which also makes me think the serial connection's not working.

I've tried many variations:

- switch tx and rx in Tasmota configuration
- switch tx and rx wiring connections
- add a level shifter, and switch tx and rx there
- use D3 and D4 as `MiEl HVAC Tx` and rx in Tasmota (the Huzzah board I have does have builtin USB so I wondered if it could be interfering)

I'm not sure what else to try, or how to debug why my ES8266 isn't communicating with the heat pump. Would anyone have advice / recommendations for things to try? Thanks in advance!

**Comment by markfickett** [comment]
Attaching a couple photos.

First, the Huzzah ESP8266 connected via pigtail to the CN105. I think in this photo I was trying out D3/D4 instead of one of the various Tx/Rx/level-shift combos I mainly focused on.
![hvac](https://github.com/arendst/Tasmota/assets/307950/92d3e700-1c59-48b8-863d-1eaa31df87f4)

And second, a closer photo of what I believe is the CN105 connector, with the original RedLINK connector just detached.
![cn105](https://github.com/arendst/Tasmota/assets/307950/442a7a50-efd5-4a43-b38d-0f1965cd7e19)

**Comment by markfickett** [comment]
I had a Huzzah variant which had builtin USB, and that did not work. Switching to a Huzzah with no builtin USB (programmed via FTDI) everything worked fine, no level shifter needed, no explicit baud settings, same firmware build. The wiring for my case was: Red to Huzzah Tx, Black to Huzzah Rx, Yellow = 5v, Green = Gnd, Blue = N/C (12v).

Here's the board that worked, hooked up to the MSZ-GL06NA:
![huzzah](https://github.com/arendst/Tasmota/assets/307950/deb81c78-c877-4eda-9220-1cc15f8e5cd7)

And here's the board that **did not** work, along with the level shifter I tried but did not need. (Oddly I had ordered 3 of the no-USB kind and Adafruit swapped 2 of these in instead. I can't actually find them on their site, the USB Huzzah's listed have a battery plug too.)
![huzzah-no](https://github.com/arendst/Tasmota/assets/307950/1b3f3bbd-4dda-4565-b504-13c113ac1f1f)

Secondary to the Tasmota communication, but I did also get this integrated into HomeAssistant easily.

After [adding a climate entry in the HA config](https://github.com/maslyankov/Tasmota-MIEL-HVAC/blob/main/home-assistant-config.yaml), I was able to get a thermostat card:
![image](https://github.com/arendst/Tasmota/assets/307950/8fb58785-e641-415a-add4-056dab3702d3)

I actually put the `climate` config [in a package](https://www.home-assistant.io/docs/configuration/packages/) included from a separate YAML for my `climate` config, like:

`configuration.yaml`
```yaml
homeassistant:
  packages:
    minisplits:
      mqtt:
        climate: !include minisplits/mqttclimates.yaml
```

`minisplits/mqttclimates.yaml`
```yaml
- name: "Office Mini Split"
  availability_topic: ...
```

**Reply by barbudor** [reply]
On most of boards, the usb/serial will not disturb the serial if usb is not connected
If this is unfortunately the case with your Huzzah, you may try to use the alternate serial pins:
- GPIO15 for TX
- GPIO13 for RX

GPIO15 is still a sensible pin regarding boot so connecting something may prevent your board to boot (the pin has a pull down on the board, if your meter has a pull-up that will conflict and prevent the boot)
But worth a try
